Src/Base/AMReX_Algorithm.H

```diff
@@ -208,6 +208,22 @@ namespace amrex
 #endif
     }
 
+    template<typename ItType, typename ValType,
+        typename std::enable_if<std::is_floating_point<typename ItType::value_type>::value,int>::type = 0,
```

Comment:
I think we need to use `iterator_traits` otherwise raw pointers would not work as `ItType`.

We could also combine the two enable_ifs into one.

Reply:
https://en.cppreference.com/w/cpp/iterator/iterator_traits

Reply:
Thanks for the link. This version uses `iterator_traits` and combines the two `enable_if`